R/get_network.R

```diff
@@ -183,15 +192,22 @@ get_DM <- function(network, comid, distance = NULL) {
   all <- private_get_DM(network, comid)
 
   if (!is.null(distance)) {
-    stop_pathlength <- start_comid$Pathlength +
-      start_comid$LENGTHKM -
-      distance
+    stop_pathlength <- start_comid$Pathlength + start_comid$LENGTHKM - distance
 
-    network <- filter(network, COMID %in% all)
+    network <- network %>%
+      filter(COMID %in% all$COMID,
+             (Pathlength + LENGTHKM) >= stop_pathlength,
+             COMID != comid) %>%
+      arrange(desc(Pathlength))
+
+    return(network$COMID)
 
-    return(filter(network, (Pathlength + LENGTHKM) >= stop_pathlength)$COMID)
   } else {
-    return(all)
+
+    all <- all %>%
+      arrange(desc(Pathlength)) %>%
+      filter(COMID != comid)
```

Comment:
I prefer having these navigations be inclusive of the starting catchment. Is there a strong argument to exclude it?

Reply:
For all use cases I've had (looking for related flow values and geometries in a NWM context) I am truly looking for upstream (or downstream) reaches. The inclusion of the input reach requires filtering it out, outside of the function, and was something I found myself doing everytime.

Reply:
hmmm. ok, interesting. I see how that makes sense in the context of a predictive model based on the actual catchments. You are looking for things upstream of the upstream end of a catchment or downstream of the downstream end of a catchment. The way I'm thinking about this is a bit more general... 

I've got a few instances in other work that would need to do `c(COMID, get_UM(COMID, network)` which isn't a real big deal but would be a bet of a pain. 

I guess I prefer not to change it or to add a parameter that makes it inclusive or exclusive that defaults to inclusive to be consistent with the NLDI which is inclusive. e.g. : https://cida.usgs.gov/nldi/comid/23735691/navigate/UM